### ISSUE TITLE
The `start` function returns another function, thus causing a compiler warning

### ISSUE THREAD

**Issue by fsoikin** [fable-compiler/fable-arch#53]
```
let app = createApp initModel view update renderer
start app   // warning here
```

The last line produces a compiler warning saying "This is a function value, you're probably missing an argument". This is because the `start` function returns not `unit`, but another function of type `AppMessage<_,_> -> unit`. This is apparently not an idle mistake, but is [used in DevTools](https://github.com/fable-compiler/fable-arch/blob/9338fba3f5783d6a19a7d21b44d7fd2d2f6fd12b/src/Fable.Arch/Fable.Arch.DevTools.fs#L406), as pointed out by @mastoj.

The proposed fix is to have two flavors of `start` - one returning the message intake function, and one returning `unit`:

```
// use this for devTools and other similar clever things
let startAndExposeMessageSink ... =
    // the contents of today's `start` goes here

// use this for normal apps written by mere mortals
let start ... = 
    startAndExposeMessageSink ... |> ignore
```

**Comment by MangelMaxime** [comment]
@fsoikin Your changes should be available in fable-arch 0.10.0

**Comment by fsoikin** [comment]
Thank you.
